project-koku/koku
# COST-3482:  Update hcs aws sql

## Jira Ticket

[COST-3482](https://issues.redhat.com/browse/COST-3482)

## Description

This change updates hcs aws sql to add more pattern matching cases for `lineitem_legalentity` and `lineitem_lineitemdescription`. 

## Testing

1. Testing this pr requires changes in this nise [pr](https://github.com/project-koku/nise/pull/432) to make sure you generate appropriate test data.
   - First follow the testing instructions in this nise pr and then run this command in your nise pipenv shell
```
nise report aws --static-report-file aws_static_data.yml --aws-s3-report-name None --aws-s3-bucket-name /path/to/koku/testing/local_providers/aws_local
```
(Make sure to change the bucket location to your location of koku)

In koku:

2. Checkout Branch
3. add ENABLE_HCS_DEBUG=True to your env file
4. Restart Koku
5. run command `make create-test-customer`. This will populate the `aws_line_items_daily` hive table with the nise data generated in this [PR](https://github.com/project-koku/nise/pull/432).

6. Once the data is loaded, go to your trino shell by running this command:
```
trino --server localhost:8080 --catalog hive --schema org1234567 --user admin --debug
```

7. check that rows with `AWS` as value in `lineitem_legalentity` column and `$0.1158 per On Demand ("Red Hat Enterprise Linux" or "RHEL") with HA t3.small Instance Hour` in `lineitem_lineitemdescription`column, by running 

```
SELECT lineitem_usagestartdate, lineitem_legalentity, lineitem_lineitemdescription
FROM hive.org1234567.aws_line_items_daily
WHERE
    year = '2023'
    AND month = '02'
    AND (
      (
        lineitem_legalentity like '%AWS%'
        AND lineitem_lineitemdescription like '%Red Hat%'
      )
      OR (
        lineitem_legalentity like '%AWS%'
        AND lineitem_lineitemdescription like '%RHEL%'
      )
    );
```

You should see a return similar to this:
```
lineitem_usagestartdate | lineitem_legalentity |                         lineitem_lineitemdescription                          
-------------------------+----------------------+-------------------------------------------------------------------------------
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand RHEL with HA t3.small Instance Hour                     
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand Red Hat Enterprise Linux with HA t3.small Instance Hour 
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand Red Hat Enterprise Linux with HA t3.small Instance Hour 
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand Red Hat Enterprise Linux with HA t3.small Instance Hour 
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand Red Hat Enterprise Linux with HA t3.small Instance Hour 
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand Red Hat Enterprise Linux with HA t3.small Instance Hour 
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand Red Hat Enterprise Linux with HA t3.small Instance Hour 
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand Red Hat Enterprise Linux with HA t3.small Instance Hour 
 2023-02-26 00:00:00.000 | AWS                  | $0.1158 per On Demand Red Hat Enterprise Linux with HA t3.small Instance Hour 
```
